app/models/extended_metadata_type.rb

```diff
@@ -62,6 +52,16 @@ def cannot_disable_nested_extended_metadata
     end
   end
 
+  def supports_extended_metadata
+    begin
+      unless self.supported_type.constantize.supports_extended_metadata?
```

Comment:
use `Seek::Util.lookup_class` instead of `constantize` as it's safer and more secure